sub-packages/bionemo-scdl/src/bionemo/scdl/io/single_cell_memmap_dataset.py

```diff
@@ -611,14 +611,15 @@ def load_h5ad(
         file_size_MB = os.path.getsize(anndata_path) / (1_024**2)
 
         if file_size_MB < self.paginated_load_cutoff:
-            features, num_rows = self.regular_load_h5ad(anndata_path)
+            features_df, num_rows = self.regular_load_h5ad(anndata_path)
 
         else:
-            features, num_rows = self.paginated_load_h5ad(anndata_path)
-
-        # Collect features and store in FeatureIndex
-        self._feature_index.append_features(n_obs=num_rows, features=features, label=anndata_path)
+            features_df, num_rows = self.paginated_load_h5ad(anndata_path)
 
+        features = {col: np.array(features_df[col].values) for col in features_df.columns}
+        self._feature_index.append_features(
+            n_obs=num_rows, features=features, num_genes=len(features[next(iter(features.keys()))]), label=anndata_path
+        )
```

Comment:
Is there an easier way write this than len(features[next(iter(features.keys()))]? 
Maybe outside of this, define number_genes = [len(features[key] )for key in features]